concept/type_/entity_type.rs

```diff
@@ -94,12 +94,16 @@ impl<'a> TypeAPI<'a> for EntityType<'a> {
         snapshot: &impl ReadableSnapshot,
         type_manager: &TypeManager,
     ) -> Result<bool, ConceptReadError> {
-        let annotations = self.get_annotations(snapshot, type_manager)?;
-        Ok(annotations.contains_key(&EntityTypeAnnotation::Abstract(AnnotationAbstract)))
+        type_manager.get_type_is_abstract(snapshot, self.clone())
     }
 
-    fn delete(self, snapshot: &mut impl WritableSnapshot, type_manager: &TypeManager) -> Result<(), ConceptWriteError> {
-        type_manager.delete_entity_type(snapshot, self.clone().into_owned())
+    fn delete(
+        self,
+        snapshot: &mut impl WritableSnapshot,
+        type_manager: &TypeManager,
+        thing_manager: &ThingManager,
+    ) -> Result<(), ConceptWriteError> {
+        type_manager.delete_entity_type(snapshot, thing_manager, self.clone().into_owned())
```

Comment:
Not sure we need the `.clone()` on self since you have self by ownership!

Reply:
Gp!